tests/attacks/evasion/test_adversarial_texture_pytorch.py

```diff
@@ -0,0 +1,199 @@
+# MIT License
+#
+# Copyright (C) The Adversarial Robustness Toolbox (ART) Authors 2021
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
+# documentation files (the "Software"), to deal in the Software without restriction, including without limitation the
+# rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit
+# persons to whom the Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all copies or substantial portions of the
+# Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
+# WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+# TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+import os
+import logging
+
+import numpy as np
+import pytest
+
+from art.attacks.evasion import AdversarialTexturePyTorch
+from art.estimators.estimator import BaseEstimator, LossGradientsMixin
+from art.estimators.object_tracking.object_tracker import ObjectTrackerMixin
+from art.estimators.object_tracking import PyTorchGoturn
+
+from tests.attacks.utils import backend_test_classifier_type_check_fail
+from tests.utils import ARTTestException
+
+logger = logging.getLogger(__name__)
+
+
+@pytest.fixture()
+def fix_get_mnist_subset(get_mnist_dataset):
+    (x_train_mnist, y_train_mnist), (x_test_mnist, y_test_mnist) = get_mnist_dataset
+    n_train = 10
+    n_test = 10
+    yield x_train_mnist[:n_train], y_train_mnist[:n_train], x_test_mnist[:n_test], y_test_mnist[:n_test]
+
+
+@pytest.mark.skip_module("scripts")
+@pytest.mark.skip_framework("tensorflow", "keras", "kerastf", "mxnet", "non_dl_frameworks")
+def test_generate(art_warning, fix_get_mnist_subset, fix_get_goturn, framework):
+    try:
+        import torch
+        from scripts.train import GoturnTrain
+        from pathlib import Path
+
+        _device = "cpu"
```

Comment:
If a GPU is available, would it make sense to run the tests on a GPU to ensure that there is device consistency on this and future modifications of the code?

Reply:
That's good idea and would be very nice, at the moment there are no GPUs available on Actions. At the moment we run the tests manually on a GPU.